In [3]:
import pandas as pd
import numpy as np
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, roc_auc_score
import matplotlib.pyplot as plt
import seaborn as sns

In [13]:
df = pd.read_csv('results/dla_results.csv')

In [14]:
y_true  = df['True_Label'].astype(int)
y_pred  = df['Predicted_Label'].astype(int)
y_score = df['Prediction_Probability']

In [15]:
accuracy  = accuracy_score(y_true, y_pred)
precision = precision_score(y_true, y_pred, zero_division=0)
recall    = recall_score(y_true, y_pred, zero_division=0)
f1        = f1_score(y_true, y_pred, zero_division=0)

In [16]:
np.random.seed(42)
n_bootstraps = 100
auc_scores   = []

In [17]:
for _ in range(n_bootstraps):
    indices      = np.random.choice(len(y_true), size=len(y_true), replace=True)
    y_true_boot  = y_true.iloc[indices]
    y_score_boot = y_score.iloc[indices]
    auc          = roc_auc_score(y_true_boot, y_score_boot)
    auc_scores.append(auc)

In [18]:
mean_auc = np.mean(auc_scores)
ci_lower = np.percentile(auc_scores, 2.5)
ci_upper = np.percentile(auc_scores, 97.5)

In [19]:
print(f"Accuracy: {accuracy:.4f}")
print(f"Precision: {precision:.4f}")
print(f"Recall: {recall:.4f}")
print(f"F1 Score: {f1:.4f}")
print(f"\nBootstrapped AUC (mean): {mean_auc:.4f}")
print(f"95% Confidence Interval: ({ci_lower:.4f}, {ci_upper:.4f})")

Accuracy: 0.9705
Precision: 0.9513
Recall: 0.9965
F1 Score: 0.9734

Bootstrapped AUC (mean): 0.9978
95% Confidence Interval: (0.9962, 0.9993)


Accuracy: 0.9696
Precision: 0.9482
Recall: 0.9982
F1 Score: 0.9726

Bootstrapped AUC (mean): 0.9983
95% Confidence Interval: (0.9973, 0.9991)

Accuracy: 0.9705
Precision: 0.9513
Recall: 0.9965
F1 Score: 0.9734

Bootstrapped AUC (mean): 0.9978
95% Confidence Interval: (0.9962, 0.9993)